assets/src/blocks/Media/MediaEditor.js

```diff
@@ -0,0 +1,143 @@
+
+  function onSelectImage({id}) {
+    setAttributes({video_poster_img: id});
+  }
+
+  function onUploadError({message}) {
+    console.log(message);
+  }
+
+  return (
+    <Fragment>
+      <header>
+        <RichText
+          tagName="h2"
+          className="page-section-header"
+          placeholder={__('Enter title', 'planet4-blocks-backend')}
+          value={video_title}
+          onChange={toAttribute('video_title')}
+          keepPlaceholderOnFocus={true}
+          withoutInteractiveFormatting
+          characterLimit={40}
+          multiline="false"
+        />
+      </header>
+      <RichText
+        tagName="p"
+        className="page-section-description"
+        placeholder={__('Enter description', 'planet4-blocks-backend')}
+        value={description}
+        onChange={toAttribute('description')}
+        keepPlaceholderOnFocus={true}
+        withoutInteractiveFormatting
+        characterLimit={200}
+      />
+      <MediaPlaceholder
+        labels={{ title: __('Video poster image [Optional]', 'planet4-blocks-backend'), instructions: __('Applicable for .mp4 video URLs only.', 'planet4-blocks-backend')}}
+        icon="format-image"
+        onSelect={ onSelectImage }
+        onError={ onUploadError }
+        accept="image/*"
+        allowedTypes={["image"]}
+      />
+      {
+        (youtube_id && !youtube_id.endsWith('.mp4')) && !embedHtml
+        ? <div className="block-edit-mode-warning components-notice is-error">
+            { __( 'The video URL could not be parsed.', 'planet4-blocks-backend' ) }
+          </div>
+        : <Fragment>
+            {
+              youtube_id && youtube_id.endsWith('.mp4')
+              ? <MediaElementVideo videoURL={ youtube_id } videoPoster={ posterURL } />
+              : <MediaEmbedPreview html={ embedHtml || null } />
+            }
+          </Fragment>
+      }
+    </Fragment>
+  )
+}
+
+export const MediaEditor = (props) => {
+  const { attributes, setAttributes, isSelected } = props;
+  const { youtube_id, video_poster_img } = attributes;
+  let mediaURL = youtube_id;
+
+  const toAttribute = attributeName => value => setAttributes({ [attributeName]: value });
+
+  // Assume that a non-URL is a YouTube video ID, for back compat.
+  if ( youtube_id && youtube_id.indexOf('/') === -1 ) {
+    mediaURL = `https://www.youtube.com/watch?v=${mediaURL}`;
+  }
+
+  if ( youtube_id && youtube_id.indexOf('youtube.com') > -1 ) {
+    mediaURL.replace('youtube.com', 'youtube-nocookie.com');
+  }
+
+  const { embedHtml, posterURL } = useSelect((select) => {
+    const embedPreview = select('core').getEmbedPreview( mediaURL );
+    const media = select('core').getMedia(video_poster_img)
+    return {
+      embedHtml: embedPreview ? embedPreview.html : null,
+      posterURL: media ? media.media_details.sizes.large.source_url : null
+    }
+  },
+  [ youtube_id, video_poster_img ]);
+
+  useEffect(() => {
+    setAttributes({
```

Comment:
This effect is part of the "static render" path:

In order to render the block statically with no endpoint calls we need to store the embed's HTML and the poster image URL.

As that content is fetched when the Media URL or the image ID changes, the effect itself is more or less a change listener for ` embedHTML` and `posterURL`, which are retrieved in the previous `useSelect` call right before this effect. Once we know the content of `embedHTML` and `posterURL`, we store them as attributes for the block to render statically. 

I'll add this as a comment to the code.

Reply:
Hmm no, I don't think so, can you come up with a test case?

Reply:
Well it's effectively a component modifying its own props in render. But indeed in its current form the code won't go in a loop because the effect ensures it only modifies 2 certain props right after 2 other props have changed.

Reply:
Shouldn't we ensure instead that the `embed_html` prop gets set whenever the `media_url` prop gets set?

Reply:
Or is this to account for old content not having the `embed_html` prop yet?

Reply:
Hmmm not sure I follow, `embed_html` and `poster_url` are actually local in this scope, so `embed_html` is not obtained from `attributes`, it's the result of the `useSelect` call above which is fired when `media_url` changes. 

This is not happening in render, it's inside an effect. And the effect's dependencies lists exactly the values it should check to avoid unnecessary re-renders.. 

What is the real risk here? We could discuss every line on a PR but we need to find closure at some point, this has 115 comments already. 💥

Reply:
I mean, re-renders caused by the attribute being set over and over by the effect.

Reply:
This was superseded by the merged PR. https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/423